oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/ElasticsearchClientFactory.java

```diff
@@ -25,32 +24,36 @@
 
 import java.io.Closeable;
 import java.io.IOException;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
-import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
-public class ElasticsearchConnectionFactory implements Closeable {
-    private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchConnectionFactory.class);
-    private final ConcurrentMap<ElasticsearchCoordinate, RestHighLevelClient> clientMap = Maps.newConcurrentMap();
+public class ElasticsearchClientFactory implements Closeable {
+    private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchClientFactory.class);
 
+    public static final ElasticsearchClientFactory INSTANCE = new ElasticsearchClientFactory();
```

Comment:
Does it need to be public, if getInstance is public?

Reply:
done